### PR TITLE
Preparations for removing `ReporterInput.resolutionProvider`

### PR DESCRIPTION
**Pull request by fviernau**
See individual commits.

### CODE REVIEW

File: helper-cli/src/main/kotlin/commands/repoconfig/GenerateRuleViolationResolutionsCommand.kt
Comment (mnonnenmacher) [inline helper-cli/src/main/kotlin/commands/repoconfig/GenerateRuleViolationResolutionsCommand.kt:1]: Commit message: This is a breaking API change. (nit: and maybe "remove" instead of "eliminate")

File: model/src/main/kotlin/OrtResult.kt
Comment (mnonnenmacher) [inline model/src/main/kotlin/OrtResult.kt:1]: Commit message: Should now be changed to say "...to the resolved resolutions instead of those from the repository configuration".

File: plugins/commands/reporter/src/main/kotlin/ReporterCommand.kt
Comment (mnonnenmacher) [inline plugins/commands/reporter/src/main/kotlin/ReporterCommand.kt:1]: Commit message: "ResolutionResolver" -> "ResolutionProvider"

File: plugins/commands/reporter/src/main/kotlin/ReporterCommand.kt [hunk 1]
```diff
@@ -204,10 +205,10 @@ class ReporterCommand : OrtCommand(
             ortResult = ortResult.replaceConfig(config)
         }
 
-        val resolutionProvider =
-            ortResult.resolvedConfiguration.resolutions.takeUnless { refreshResolutions }?.let { resolutions ->
-                DefaultResolutionProvider(resolutions)
-            } ?: DefaultResolutionProvider.create(ortResult, resolutionsFile)
+        if (refreshResolutions) {
```
Comment (mnonnenmacher) [inline plugins/commands/reporter/src/main/kotlin/ReporterCommand.kt:208]: Condition should be `refreshResolutions || ortResult.resolvedConfiguration.resolutions == null` to also resolve resolutions if they have not been resolved before (becuase the evaluator did not run).

File: plugins/reporters/evaluated-model/src/funTest/assets/reporter-test-input.yml
Comment (mnonnenmacher) [inline plugins/reporters/evaluated-model/src/funTest/assets/reporter-test-input.yml:1]: Commit message:
* First line "resolution" -> "resolutions"
* "included in **the** `OrtResult` before passing it ~~around~~ to the reporters"

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: helper-cli/src/main/kotlin/commands/repoconfig/GenerateRuleViolationResolutionsCommand.kt
Comment (fviernau) [reply helper-cli/src/main/kotlin/commands/repoconfig/GenerateRuleViolationResolutionsCommand.kt:1]: Eliminated eliminate as you prefer.

File: model/src/main/kotlin/OrtResult.kt
Comment (fviernau) [reply model/src/main/kotlin/OrtResult.kt:1]: The commit message was phrased how @sschuberth asked it to be. I don't understand the rationale why out should change now again, in particular because the commit diff hasn't change in the meantime. Please provide a rationale, and if you insist on the change it would be nice to have the exact sentence you want to see, because from the above I am unable to guess what your expectations are.

File: plugins/commands/reporter/src/main/kotlin/ReporterCommand.kt (hunk 1 above)
Comment (fviernau) [reply plugins/commands/reporter/src/main/kotlin/ReporterCommand.kt:208]: Nice catch.

File: model/src/main/kotlin/OrtResult.kt
Comment (mnonnenmacher) [reply model/src/main/kotlin/OrtResult.kt:1]: But didn't that discussion refer to the version of `getResolutions()` where it returned a merge of the resolutions in the repository config and the resolved config? As it now only returns those from the resolved config, I would also only refer to that. For example:

"Make the functions adhere to the resolutions from the resolved configuration instead of those from the repository configuration."

Comment (sschuberth) [reply model/src/main/kotlin/OrtResult.kt:1]: > But didn't that discussion refer to the version of `getResolutions()` where it returned a merge of the resolutions in the repository config and the resolved config?

Indeed it did.

Comment (fviernau) [reply model/src/main/kotlin/OrtResult.kt:1]: I've fixed it.

File: model/src/test/kotlin/OrtResultTest.kt [hunk 2]
```diff
@@ -146,29 +146,26 @@ class OrtResultTest : WordSpec({
...
                     result = AnalyzerResult(
                         projects = emptySet(),
                         packages = emptySet(),
                         issues = mapOf(
-                            Identifier("Maven:org.oss-review-toolkit:example:1.0") to
-                                listOf(
-                                    Issue(message = "Issue message to resolve", source = ""),
-                                    Issue(message = "Non-resolved issue", source = "")
-                                )
+                            Identifier("Maven:org.oss-review-toolkit:example:1.0") to listOf(
```
Comment (sschuberth) [inline model/src/test/kotlin/OrtResultTest.kt:154]: Just for my understanding: Is this change necessary because the original resolution from the repo config was not actually "baked" into the resolutions from the resolved configuration in the hard-coded test asset, although it usually *would* be if the repo config is read by ORT?

Comment (sschuberth) [reply model/src/test/kotlin/OrtResultTest.kt:154]: I.e. this is a bit similar to the adjustments of the test asset in your last commit 398a6215568581f12d567ed362aab1ed3834d68f, correct?

**Review by mnonnenmacher — APPROVED** [review]
